### PR TITLE
Misc cleanup, to reduce diff vs. upstream.

### PR DESCRIPTION
**Pull request by hlinnaka**

### CODE REVIEW

File: src/backend/cdb/cdbpartition.c [hunk 1]
```diff
@@ -497,48 +497,6 @@ rel_has_external_partition(Oid relid)
 }
 
 /*
- * Check if a Query struct has external partition relation
- */
-bool
-query_has_external_partition(Query *query)
```
Comment (vraghavan78) [inline src/backend/cdb/cdbpartition.c]: isn't this used in cdbpartition?

Comment (vraghavan78) [inline src/backend/cdb/cdbpartition.c]: is there a different func?

File: src/backend/cdb/cdbpartition.c [hunk 2]
```diff
@@ -450,8 +450,17 @@ rel_partition_keys_kinds_ordered(Oid relid, List **pkeys, List **pkinds)
 	// now order the keys and kinds by level
 	for (int i = 0; i< nlevels; i++)
 	{
-		int pos = list_find_int(levels, i);
-		Assert (0 <= pos);
+		ListCell   *cell;
+		int			pos = 0;
+
+		/* Find i's position in the 'levels' list. */
+		foreach(cell, levels)
```
Comment (ashwinstar) [inline src/backend/cdb/cdbpartition.c]: Can we use `list_nth_int()` here ?

File: src/backend/cdb/cdbpartition.c (hunk 1 above)
Comment (hlinnaka) [inline src/backend/cdb/cdbpartition.c]: There is `rel_has_external_partition`, which is used, among other things, in ORCA to bail out `query_has_external_partition` is indeed unused.

File: src/backend/cdb/cdbpartition.c (hunk 2 above)
Comment (hlinnaka) [inline src/backend/cdb/cdbpartition.c]: `list_nth_int` returns the nth element of the list. This is the opposite: it finds the given element in the list, and returns its position.

**Comment by hlinnaka** [comment]
Pushed, thanks for the reviews.

**Pull request closed without merging**